packages/animations/browser/src/render/web_animations/web_animations_player.ts

```
@@ -155,14 +159,20 @@ export class WebAnimationsPlayer implements AnimationPlayer {
   }
 
   setPosition(p: number): void {
-    if (this.domPlayer === undefined) {
+    if (this._domPlayer === undefined) {
       this.init();
     }
-    this.domPlayer.currentTime = p * this.time;
+    // init initialized the _domPlayer so
+    // the non-null assertion can be safely used here
+    this._domPlayer!.currentTime = p * this.time;
   }
 
   getPosition(): number {
-    return this.domPlayer.currentTime / this.time;
+    if (this._domPlayer === undefined) {
```

The current logic will cause a JS exception in case `this.domPlayer` is undefined. I think we can retain that behavior, just make the error message better (and use `ngDevMode` check for tree-shaking):

```
getPosition(): number {
  if (ngDevMode && this._domPlayer === undefined) {
    throw new Error(' ... ');
  }
  return this._domPlayer.currentTime / this.time;
}
```

---

ah yes I can definitely do that :slightly_smiling_face: as I commented below I do think that the function is currently never called when the domPlayer is not initialized so I think would be fine to change the behaviour of such situation a bit

but yeah probably this would be the best approach rather than returning `NaN` which could cause subsequent code to fail in ways potentially much more cryptic and difficult to diagnose :slightly_smiling_face::+1: